Crash Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x1333620), concrete:(*runtime._type)(0x14c5b00), asserted:(*runtime._type)(0x12f5da0), missingMethod:""} (interface conversion: interface {} is *v1.Route, not string)

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via email to: tnozicka /at/ gmail /dot/ com .
-->


**What happened**:
acme pod is crashloopbacking

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
`
I0525 07:26:20.568812       1 route.go:495] Finished syncing Route "automatic-test/secure-keycloak"
E0525 07:26:31.841969       1 runtime.go:78] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x1333620), concrete:(*runtime._type)(0x14c5b00), asserted:(*runtime._type)(0x12f5da0), missingMethod:""} (interface conversion: interface {} is *v1.Route, not string)
goroutine 232 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1366900, 0xc0003583f0)
	k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1366900, 0xc0003583f0)
	runtime/panic.go:679 +0x1b2
github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).processNextRouteItem(0xc0001aa5b0, 0x1714aa0, 0xc000354000, 0x306b6c6532705100)
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1293 +0x2c4
github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).runRouteWorker(0xc0001aa5b0, 0x1714aa0, 0xc000354000)
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1325 +0x3f
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:183 +0x37
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000683f08)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000895f08, 0x3b9aca00, 0x0, 0x1, 0xc000356a20)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x1714aa0, 0xc000354000, 0xc000683fa0, 0x3b9aca00, 0x0, 0x1)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:183 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:97
github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).Run.func2(0xc000046550, 0x1714aa0, 0xc000354000, 0xc0001aa5b0)
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1357 +0xb4
created by github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).Run
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1355 +0x22a
panic: interface conversion: interface {} is *v1.Route, not string [recovered]
	panic: interface conversion: interface {} is *v1.Route, not string

goroutine 232 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1366900, 0xc0003583f0)
	runtime/panic.go:679 +0x1b2
github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).processNextRouteItem(0xc0001aa5b0, 0x1714aa0, 0xc000354000, 0x306b6c6532705100)
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1293 +0x2c4
github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).runRouteWorker(0xc0001aa5b0, 0x1714aa0, 0xc000354000)
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1325 +0x3f
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:183 +0x37
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000683f08)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000895f08, 0x3b9aca00, 0x0, 0x1, 0xc000356a20)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x1714aa0, 0xc000354000, 0xc000683fa0, 0x3b9aca00, 0x0, 0x1)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:183 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)
	k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:97
github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).Run.func2(0xc000046550, 0x1714aa0, 0xc000354000, 0xc0001aa5b0)
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1357 +0xb4
created by github.com/tnozicka/openshift-acme/pkg/controller/route.(*RouteController).Run
	github.com/tnozicka/openshift-acme@/pkg/controller/route/route.go:1355 +0x22a

`

**Environment**:
- OpenShift/Kubernetes version (use `oc/kubectl version`):
- Others:


@tnozicka

--- a/pkg/controller/route/route.go
+++ b/pkg/controller/route/route.go
@@ -1049,7 +1049,7 @@ func (rc *RouteController) sync(ctx context.Context, key string) error {
 
 	case acme.StatusProcessing:
 		// TODO: backoff but capped at some reasonable time
-		rc.queue.AddAfter(routeReadOnly, 15*time.Second)
+		rc.queue.AddAfter(key, 15*time.Second)
 
 		klog.V(4).Infof("Route %q: Order %q: Waiting to be validated by ACME server", key, order)
 

